README.md

```diff
@@ -274,189 +283,566 @@ This is defined in libocpp/include/ocpp/v16/charge_point.hpp and takes the follo
+- `on_transaction_stopped`
 
   Notifies libocpp that the transaction on the given connector with the given reason has been stopped.
 
   Some of its parameters:
 
-  timestamp at the end of the transaction
-
-  energy_wh_import contains the meter value in Wh for the connector at end of the transaction
+  - `timestamp` at the end of the transaction
+  - `energy_wh_import` contains the meter value in Wh for the connector at end of the transaction
 
-- on_suspend_charging_ev
+- `on_suspend_charging_ev`
 
   Notifies libocpp that the EV has paused charging
 
-- on_suspend_charging_evse
+- `on_suspend_charging_evse`
 
   Notifies libocpp that the EVSE has paused charging
 
-- on_resume_charging
+- `on_resume_charging`
 
   Notifies libocpp that charging has resumed
 
-- on_session_started
+- `on_session_started`
 
   this is mostly used for logging and changing the connector state
 
-- on_session_stopped
+- `on_session_stopped`
 
   this is mostly used for logging and changing the connector state
 
-- on_error
+- `on_error`
 
   Notify libocpp of an error
 
-- on_reservation_start
+- `on_reservation_start`
 
   Notifies libocpp that a reservation has started
 
-- on_reservation_end
+- `on_reservation_end`
 
   Notifies libocpp that a reservation has ended
 
 #### Authorization
-In EVerest authorization is handled by the Auth module and various auth token providers and validators. The OCPP module acts as both a token provider (for pre validated tokens in RemoteStartTransactions) and a token validator (using the authorize requests, or plug & charge)
-To use libocpp as a auth token validator (e.g. before starting a transaction) you can call the "authorize_id_token" function of the ChargePoint object
 
-## Integrate this library with your Charging Station Implementation for OCPP2.0.1
-TODO
+In EVerest authorization is handled by the Auth module and various auth token providers and validators. The OCPP module acts as both a token provider (for pre validated tokens in RemoteStartTransactions) and a token validator (using the authorize requests, or plug & charge).
+
+To use libocpp as a auth token validator (e.g. before starting a transaction) you can call the "authorize_id_token" function of the ChargePoint object.
+
+## Usage for OCPP 2.0.1
+
+### Smart Charging
+
+Work to fully support OCPP 2.0.1 Smart Charging is ongoing. Most functional requirements for General Smart Charging use cases (that is, K01–K10) are now supported. For an up-to-date overview of which features are currently supported, please refer to the [OCPP 2.0.1 Status](doc/ocpp_201_status.md) document.
+
+#### K01 SetChargingProfile
+
+Allows the CSMS to influence the charging power or current drawn from a specific EVSE or the
+entire Charging Station over a period of time.
+
+```mermaid
+sequenceDiagram
+    CSMS->>+ChargePoint : SetChargingProfileRequest(call)
+
+    ChargePoint->>+DeviceModel : SmartChargingCtrlrAvailable?
+    DeviceModel-->>-ChargePoint : Component
```

Comment:
nice! Maybe @nserway should use this instead of PlantUML Having said that though, I think that the README should not have this level of detail. This OCPP 2.0.1 architecture diagram and documentation should be moved into a separate document.

Reply:
@shankari do you have a preference on the location? My thought is to split it out into a markdown of its own in the `doc` directory and then link to it in the overall file.

Reply:
Per a conversation on the PR the above move was accepted.